UI/contact-list.ts

```diff
@@ -61,9 +62,23 @@ export class ProfilesSyncer {
     }
 
     async add(...users: string[]) {
-        const err = await this.chan.put(users);
-        if (err instanceof Error) {
-            throw err; // impossible
+        const needsToAdd = users.filter((user) => {
+            if (!this.syncedProfiles.has(user)) {
+                this.syncedProfiles.add(user);
+                return true;
+            }
+
+            return false;
+        });
+        if (needsToAdd.length > 0) {
+            const err = await this.chan.put(users);
```

Comment:
even your unnecessary code is buggy here.
Why not just `needsToAdd`?

But this class doesn't need to change anyway.